opsimulator/opsimulator.go

```diff
@@ -188,90 +189,57 @@ func (opSim *OpSimulator) startBackgroundTasks() {
 			case <-opSim.bgTasksCtx.Done():
 				sub.Unsubscribe()
 				close(depositTxCh)
+				return nil
 			}
 		}
 	})
 
-	// Store and log L2 to L2 messages
+	// Log L2ToL2CrossChainMessenger events
```

Comment:
Any reason the opsim needs reference to the indexer anymore if it's pulled out?

It feels like this logging could just happen within the indexer and we wouldn't need to subscriptions and stuff. wydt?

Reply:
lol yeah that should make it much cleaner. will keep the subscriptions around for the relayer part but agree the logging should happen inside the indexer